cli/actions/run_test_action.go

```diff
@@ -79,26 +80,27 @@ func (a runTestAction) runDefinition(ctx context.Context, definitionFile string,
 
 	var test openapi.Test
 
-	if definition.Id == "" {
-		a.logger.Debug("test doesn't exist. Creating it")
-		test, err = a.createTestFromDefinition(ctx, definition)
-		if err != nil {
-			return runTestOutput{}, fmt.Errorf("could not create test from definition: %w", err)
-		}
+	a.logger.Debug("try to create test")
 
-		definition.Id = *test.Id
-		err = file.SaveDefinition(definitionFile, definition)
-		if err != nil {
-			return runTestOutput{}, fmt.Errorf("could not save definition: %w", err)
-		}
-	} else {
+	test, exists, err := a.createTestFromDefinition(ctx, definition)
+	if err != nil {
+		return runTestOutput{}, fmt.Errorf("could not create test from definition: %w", err)
+	}
```

Comment:
Instead of changing how `createTestFromDefinition` works, why don't you call `a.client.ApiApi.GetTest(ctx, id)`. If it returns an empty test, you create a new one. If not, you call `updateTestFromDefinition`.

Reply:
I think this approach is more flexible, because we are not duplicating the logic of verifying if a test exists. If at some point we decide to change that logic to check things besides the ID, this method wouldn't work anymore and we'd need to update both the client and the server